apps/backend/src/user/user.service.ts

```diff
@@ -1,235 +1,245 @@
+      }
+
+      // Make sure commands aren't empty
+      if (commands.length === 0) {
+        throw new Error('No fields to update');
+      }
+
+      // Combine into one update expression and update the user
+      const updateExpr = `SET ${commands.join(', ')}`;
+      const key = { userId: { N: userId.toString() } };
+      const data = await this.dynamoDbService.updateItemWithExpression(
+        this.tableName,
+        key,
+        updateExpr,
+        expressionAttributeValues,
+        expressionAttributeNames,
+      );
+
+      const result = await this.getUser(userId);
+
+      if (!data) {
+        return { statusCode: 400, message: 'No user found with id: ' + userId };
+      }
+      return result;
+    } catch (e) {
+      throw new Error(
+        `Error updating data for user with id: ${userId}: ${e.message}`,
+      );
     }
-    /**
-     * Maps a user's data from DynamoDB to a UserModel object.
-     * @param data the user's data from DynamoDB
-     * @returns the user's information as a UserModel object
-     */
-    private mapDynamoDBItemToUserModelV2(data: { [key: string]: any }): UserModel {
-        return {
-            userId: data["userId"].N,
-            firstName: data["firstName"].S,
-            lastName: data["lastName"].S,
-            email: data["email"].S,
-            phoneNumber: parseInt(data["phoneNumber"].N),
-            siteIds: data["siteIds"]?.NS?.map(Number) ?? [],
-            zipCode: parseInt(data["zipCode"].N),
-            birthDate: new Date(data["birthDate"].S),
-            role: data["role"].S,
-            status: data["status"].S as UserStatus,
-        };
+  }
+
+  // Run backend on postman please, not SwaggerUI
+  public async getUserByStatus(status: UserStatus): Promise<UserModel[]> {
+    try {
+      const filterExpression = '#user_status = :statusOf';
+      const expressionAttributeValues = { ':statusOf': { S: status } };
+      const expressionAttributeNames = { '#user_status': 'status' };
+
+      const data = await this.dynamoDbService.scanTable(
+        this.tableName,
+        filterExpression,
+        expressionAttributeValues,
+        expressionAttributeNames,
+      );
+
+      return data.map((item) => this.mapDynamoDBItemToUserModelV2(item)); // added data
+    } catch (error) {
+      console.error('Error fetching users by status:', error);
+      throw new Error(`Error fetching users by status: ${error.message}`);
     }
-
-
-
-
-}
\ No newline at end of file
+  }
+
+  /**
+   * Maps a user's data from DynamoDB to a UserModel object.
+   * @param objectId the user's id
+   * @param data the user's data from DynamoDB
+   * @throws Error if the user's status or role is invalid
+   * @returns the user's information as a UserModel object
+   */
+  private mapDynamoDBItemToUserModel(
+    objectId: number,
+    data?: { [key: string]: any },
+  ): UserModel {
+    const siteIds = data['siteIds'].L.map((item) => Number(item.N)) ?? [];
```

Comment:
```suggestion
    const siteIds = Array.isArray(data["siteIds"]?.L) 
        ? data["siteIds"].L.map(item => Number(item.N)) 
        : [];
```
You might have to make this change to remove this error: `Cannot read properties of undefined (reading 'map')`